## Read pixel

### Read image first

In [26]:
import cv2
path = "I:\\openCV\\test.jpg"
# be careful of path and \\
image = cv2.imread(path,flags=0)
# check if the image load successfully or not 
if image is None:
    print("Fail to load")
else:
    cv2.imshow('image', image)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

### Check the grayscale value

In [17]:
p = image[88,152] #第88行，142列的值
p

172

Return value is 172. The grayscale usually start from 0 to 255, 0 means black, and 255 means white.

### Check BGR value.
#### read image first

In [18]:
image1 = cv2.imread(path) # no flag means BGR image
# check if the image load successfully or not 
if image1 is None:
    print("Fail to load")
else:
    cv2.imshow('image', image1)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

In [19]:
blue = image1[78,125,0] # the 3rd value is color tune
blue

158

In [20]:
green = image1[78,125,1]
green

169

In [21]:
red = image1[78,125,2]
red

167

In [22]:
value = image1[78,125]
value # the return value is 3 color value, order by BGR

array([158, 169, 167], dtype=uint8)

## Change pixel

In [24]:
image[88,99] = 255
cv2.imshow('image', image)
cv2.waitKey(0)
cv2.destroyAllWindows()